### PR TITLE
feat(apigatewayv2): HttpRouteIntegration supports AWS services integrations

### PR DESCRIPTION
**Pull request by jogold**
Add support for integration subtype and credentials allowing to extend
`HttpRouteIntegration` to create integrations for AWS services.

See https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-aws-services.html

Extracted part of #16287 to make it more reviewer friendly.

BREAKING CHANGE: `HttpIntegrationType.LAMBDA_PROXY` has been renamed to `HttpIntegrationType.AWS_PROXY`

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jogold** [comment]
Needs `pr-linter/exempt-readme`

File: packages/@aws-cdk/aws-apigatewayv2/lib/http/integration.ts [hunk 1]
```diff
@@ -29,9 +30,83 @@ export enum HttpIntegrationType {
...
    * Integration type is an HTTP proxy
-   * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+   * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-http.html
    */
   HTTP_PROXY = 'HTTP_PROXY',
+  /**
+   * Integration type is an AWS proxy
+   * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-aws-services.html
+   */
+  AWS_PROXY = 'AWS_PROXY',
```
Comment (otaviomacedo) [inline packages/@aws-cdk/aws-apigatewayv2/lib/http/integration.ts:40]: Having different enum members with the same value causes problems for jsii when generating code in other languages.

Comment (jogold) [reply packages/@aws-cdk/aws-apigatewayv2/lib/http/integration.ts:40]: OK, what do you suggest?

Comment (otaviomacedo) [reply packages/@aws-cdk/aws-apigatewayv2/lib/http/integration.ts:40]: Given that this is an experimental module, we can introduce a breaking change and remove the `LAMBDA_PROXY` member.

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**